File: DownloadFiles/TGNet/Chapter05/Chapter5DemoCode/VB/SimpleTiming/SimpleTiming/Form1.vb
' To use this program, use your browser to navigate to any website and then
' retrieve the source code for that site. To do this using IE, just click the View=>Source
' menu item. Copy and paste the source code into the text box on this application
' as it runs.

Imports System.Text.RegularExpressions
Public Class Form1
    ' Note the following form properties were set:
    ' AcceptButton = button1
    ' Also the following properties for the Textbox:
    ' Anchor = top, left, bottom, right
    ' Multiline = true
    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        Dim StartTime As DateTime
        Dim EndTime As DateTime
        Dim Interval As TimeSpan
        Dim numHrefs As Int16

        StartTime = DateTime.Now 'capture current seconds

        numHrefs = FindHref(txtSource.text)

        EndTime = DateTime.Now

        lblHrefs.Text = numHrefs.ToString
        Interval = EndTime - StartTime
        lblTime.Text = Interval.ToString()
    End Sub

    Public Function FindHref(ByVal strSource As String) As Integer
        Dim r As Regex
        Dim m As Match
        Dim i As Integer

        r = New Regex("href\s*=\s*(?:""(?<1>[^""]*)""|(?<1>\S+))", _
            RegexOptions.IgnoreCase Or RegexOptions.Compiled)

        m = r.Match(strSource)
        While m.Success
            i += 1
            m = m.NextMatch()
        End While
        Return i
    End Function
End Class

File: DownloadFiles/TGNet/Chapter05/Chapter5DemoCode/C#/SimpleTiming/SimpleTiming/Form1.cs
// To use this program, use your browser to navigate to any website and then
// retrieve the source code for that site. To do this using IE, just click the View=>Source
// menu item. Copy and paste the source code into the text box on this application
// as it runs.
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using System.Text.RegularExpressions;

namespace SimpleTiming
{
    public partial class Form1 : Form
    {
        // Note the following form
        // properties were set:
        // AcceptButton = button1
        // Also the following properties for the Textbox:
        // Anchor = top, left, bottom, right
        // Multiline = true
        public Form1()
        {
            InitializeComponent();
        }

        public Int16 FindHref(string strSource)
        {
            Regex r;
            Match m;
            Int16 iCount = 0;
            r = new Regex("href\\s*=\\s*(?:\"(?<1>[^\"]*)\"|(?<1>\\S+))", RegexOptions.IgnoreCase | RegexOptions.Compiled);
            m = r.Match(strSource);
            while (m.Success)
            {
                iCount += 1;
                m = m.NextMatch();
            }
            return iCount;
        }

        private void button1_Click(object sender, EventArgs e)
        {
            DateTime StartTime;
            DateTime EndTime;
            TimeSpan Interval;
            Int16 numHrefs;
            StartTime = DateTime.Now;
            numHrefs = FindHref(txtSource.Text);
            EndTime = DateTime.Now;
            lblHrefs.Text = numHrefs.ToString();
            Interval = EndTime - StartTime;
            lblTime.Text = Interval.ToString();
            
        }

    }
}